### PR TITLE
Add our own BoringSSL copy

### PR DESCRIPTION
**Pull request by 0xTim**
As part of the migration away from OpenSSL we need to vend our own BoringSSL until Crypto fills the gaps we need. Since it adds a _ton_ of files, this is a separate PR

### CODE REVIEW

**Comment by 0xTim** [comment]
@weissi @Lukaza - would appreciate your input on taking Apache 2 code into an MIT project, I think the attribution is correct, but let me know if anything is wrong!

File: NOTICES.txt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+// Copyright (c) 2017-2020 Vapor project authors
+// Licensed under MIT
+//
+// See LICENSE for license information
+//
+// SPDX-License-Identifier: MIT
+//
+//===----------------------------------------------------------------------===//
+
+This product contains a derivation of the of the Vendor BoringSSL and Build ASM scripts
```
Comment (tanner0101) [inline NOTICES.txt]: "of the of the" typo

File: Package.swift [hunk 2]
```diff
@@ -8,6 +8,9 @@ let package = Package(
     ],
     products: [
         .library(name: "JWTKit", targets: ["JWTKit"]),
+        /* This target is used only for symbol mangling. It's added and removed automatically because it emits build warnings. MANGLE_START
+        .library(name: "CVaporJWTBoringSSL", type: .static, targets: ["CVaporJWTBoringSSL"]),
```
Comment (tanner0101) [inline Package.swift]: I'd call this `CJWTKitBoringSSL` since `JWTKit` is this package's unique namespace.

**Comment by 0xTim** [comment]
@tanner0101 cool updated the name

**Pull request merged**